src/lib.rs

```diff
@@ -7,9 +10,9 @@ use std::collections::HashMap;
 /// By default, any empty dataset will return a gini of 1.0. This may be unexpected behaviour.
 /// ```
 /// use parsnip::gini;
-/// assert_eq!(gini(&vec![0, 0, 0, 1]), 0.375);
+/// assert_eq!(gini(&vec![0_usize, 0, 0, 1]), 0.375);
 /// ```
-pub fn gini(data: &[u64]) -> f32 {
+pub fn gini<T: Unsigned + ToPrimitive>(data: &[T]) -> f32 {
```

Comment:
Updated—though I didn't need `ToPrimitive`, I am using `Ord` at the moment, because it gives sorting plus the mapping without requiring `std::hash::Hash` at the moment. I would like to handle deduplication down the road from ideally just `Eq`.

Reply:
About that: you might only be able to obtain deduplication without sorting via `Eq + Hash', but I'd say this is OK, because most types which implement `Eq` also implement `Hash` anyway (both can be derived too).